src/views/activateAccount/ActivationView.js

```diff
@@ -0,0 +1,162 @@
+              });
+            }
+            else if (data.errors[0].title === "UNKNOWN_USER_EMAIL") {
+              this.setState({
+                activationState: "UNKNOWN_USER_EMAIL_FAILURE",
+              });
+            }
+            else if (data.errors[0].title === "WRONG_ACTIVATION_CODE") {
+              this.setState({
+                activationState: "WRONG_ACTIVATION_CODE_FAILURE",
+              });
+            }
+            else {
+              this.setState({
+                activationState: "UNKNOWN_FAILURE",
+              });
+            }
+          } else {
+            this.setState({
+              activationState: "UNKNOWN_FAILURE",
+            });
+          }
+        }
+      })
+      .catch((e) =>
+        this.setState({
+          activationState: "CONNECTION_FAILURE",
+        })
+      );
+  }
+
+  getIcon = () => {
+    if (this.state.activationState == "IN_PROGRESS") {
+      return (
+        <IconContext.Provider
+          value={{ className: styles.icon + " " + styles.iconLoading }}
+        >
+          <AiOutlineLoading />
+        </IconContext.Provider>
+      );
+    } else if (this.state.activationState == "SUCCESS") {
+      return (
+        <IconContext.Provider
+          value={{ className: styles.icon + " " + styles.iconSuccess }}
+        >
+          <AiOutlineCheckCircle />
+        </IconContext.Provider>
+      );
+    }  else if (this.state.activationState == "ALREADY_ACTIVATED_WARNING") {
+      return (
+        <IconContext.Provider
+          value={{ className: styles.icon + " " + styles.iconWarning }}
+        >
+          <AiOutlineCheckCircle />
+        </IconContext.Provider>
+      );
+    } else if (
+      this.state.activationState == "UNKNOWN_USER_EMAIL_FAILURE" ||
+      this.state.activationState == "WRONG_ACTIVATION_CODE_FAILURE" || 
+      this.state.activationState == "CONNECTION_FAILURE" ||
+      this.state.activationState == "UNKNOWN_FAILURE" 
+    ) {
+      return (
+        <IconContext.Provider
+          value={{ className: styles.icon + " " + styles.iconFailure }}
+        >
+          <AiOutlineExclamationCircle />
+        </IconContext.Provider>
+      );
+    }
+  };
+
+  getTitle = () => {
+    if (this.state.activationState == "IN_PROGRESS") {
+      return "Activating account...";
+    } else if (this.state.activationState == "SUCCESS") {
+      return "Account activated";
+    } else if (this.state.activationState == "ALREADY_ACTIVATED_WARNING") {
+      return "Already activated";
+    }  else if (this.state.activationState == "UNKNOWN_USER_EMAIL_FAILURE") {
+      return "Email not found";
+    }  else if (this.state.activationState == "WRONG_ACTIVATION_CODE_FAILURE") {
+      return "Wrong activation code";
+    } else if (this.state.activationState == "CONNECTION_FAILURE") {
+      return "Connection error";
+    } else if (this.state.activationState == "UNKNOWN_FAILURE") {
+      return "Unknown failure";
```

Comment:
It's ok but I think that we can change this to more simple form ex:
`
let message = {
    "IN_PROGRESS": "Activating account...",
    "SUCCESS": "Account activated",
    ...
}
return message[this.state.activationState];
`

Reply:
wow, great idea, love it